split_to_CMYK.jsx

```
@@ -1 +1,83 @@
                cmyk[0].blendMode = BlendMode.MULTIPLY
                cmyk[1].name = layerColor [2];
                cmyk[1].blendMode = BlendMode.MULTIPLY
                cmyk[2].name = layerColor [1];
                cmyk[2].blendMode = BlendMode.MULTIPLY
                cmyk[3].name = layerColor [0];
                cmyk[3].blendMode = BlendMode.MULTIPLY
              
            }
            
            doc.activeLayer = newLayer;
        }
      } catch (e) { 
    }
}
makeLayers ()

/*function splitCyan () {
	var cyan = doc.activeChannels = [doc.channels.getByName('Cyan')];
	app.executeMenuCommand('selectall');
    
}
splitCyan()*/

/*var cyanChannel = doc.activeChannels = [doc.channels.getByName('Cyan')];
var magChannel = doc.activeChannels = [doc.channels.getByName('Magenta')];
var yelChannel = doc.activeChannels = [doc.channels.getByName('Yellow')];
var blackChannel = doc.activeChannels = [doc.channels.getByName('Black')];*/
\ No newline at end of file
+var doc = app.activeDocument;
+
+//switch color mode to get 4 channels
+var CMYK = doc.changeMode(ChangeMode.CMYK);
+
+//makes 4 layers named after the channels and adjusts blending mode
+function makeLayers() {
+  try {
+    if (app.documents.length) {
+      var layerColor = ["Cyan", "Magenta", "Yellow", "Black"]
+      var doc = activeDocument;
+      var layers = doc.artLayers;
+      var cmyk = [
+        layerA = layers.add(),
+        layerB = layers.add(),
+        layerC = layers.add(),
+        layerD = layers.add()
+      ];
+
+      if (layerColor) {
+        cmyk[0].name = layerColor[3];
+        cmyk[0].blendMode = BlendMode.NORMAL
+        cmyk[1].name = layerColor[2];
+        cmyk[1].blendMode = BlendMode.MULTIPLY
+        cmyk[2].name = layerColor[1];
+        cmyk[2].blendMode = BlendMode.MULTIPLY
+        cmyk[3].name = layerColor[0];
+        cmyk[3].blendMode = BlendMode.MULTIPLY
+
+      }
+
+      doc.activeLayer = newLayer;
+    }
+  } catch (e) {}
```

For debugging purposes you may want to change this line to:

```
} catch (e) {
  alert(e)
}
```

The `catch` statement right now is consuming any error messages and allowing the script to continue processing.  Based on some of the things I'm seeing in the `makeLayers()` method I think your script is silently failing in some places and not giving you proper notice.